tests/core/factory/test_checkpoint_factory.py

```diff
@@ -0,0 +1,238 @@
+from unittest.mock import Mock
+
+import pytest
+
+from great_expectations import set_context
+from great_expectations.checkpoint.checkpoint import Checkpoint
+from great_expectations.core.factory.checkpoint_factory import CheckpointFactory
+from great_expectations.data_context import AbstractDataContext
+from great_expectations.data_context.store.checkpoint_store import CheckpointStore
+from great_expectations.exceptions import DataContextError
+
+
+@pytest.fixture
+def checkpoint_dict():
+    return {
+        "name": "oss_test_checkpoint",
+        "expectation_checkpoint_name": "oss_test_expectation_checkpoint",
+        "validations": [
+            {
+                "name": None,
+                "id": None,
+                "expectation_checkpoint_name": "taxi.demo_pass",
+                "expectation_checkpoint_ge_cloud_id": None,
+                "batch_request": None,
+            },
+        ],
+        "action_list": [
+            {
+                "action": {"class_name": "StoreValidationResultAction"},
+                "name": "store_validation_result",
+            },
+        ],
+    }
+
+
+def _assert_checkpoint_equality(actual: Checkpoint, expected: Checkpoint):
+    # Checkpoint equality is currently defined as equality of the config
+    # TODO: We should change this to a more robust comparison (instead of memory addresses)
+    actual_config = actual.config.to_json_dict()
+    expected_config = expected.config.to_json_dict()
+    assert actual_config == expected_config
```

Comment:
Not sure how often a user will need to use `__eq__` but this feels bad right now

Reply:
Hah just left a comment above about __eq__. Why does it feel bad?

Reply:
Having to go into the config attr and then having to convert that to a dict to get equality working seems like a lot of steps - I'd imagine I could just do:
```python
assert Checkpoint("a") == Checkpoint("a") # Fails right now
```